app/controllers/tasks_controller.rb

```diff
@@ -3,32 +3,44 @@
 class TasksController < ApplicationController
   before_action :set_task, only: %i[show edit update destroy]
 
+  before_action :require_signed_in!
+  before_action :check_list_owner
+
   # GET /tasks
   # GET /tasks.json
   def index
-    @tasks = Task.all
+    comp_status = params[:completed] == "true" ? true : false
```

Comment:
Do you need the ternary here?

Reply:
I did this to keep the code concise rather than doing a longer `if` statement. This code's purpose is to check if a query was sent with the get request address. In my code, on the view page for the index, a user can click on a link that then adds a query to the end of the address `/lists/1/tasks?completed=true` this would then trigger `comp_status` (which is by default  `false` since `params[:completed]` when no query exists would be a falsey value) to be `true` which would in turn only display tasks that are marked as completed.

Reply:
However, it would be completely valid to use a `if else` statement in this case also

Reply:
I see, thank you for the clarification. The reason the ternary is there is that I was thinking that I could maybe expand on it later by adding additional cases rather than a true boolean. So if I decided to add a link that would allow all tasks to be viewed in addition to just completed or not completed, I could have add additional conditions to the ternary.